src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/MovementServiceTest.kt

```diff
@@ -49,26 +70,93 @@ class MovementServiceTest {
+  }
+
+  @Test
+  fun `getMovementsMatchedWithPrisoner - adds PNC from Prison Number match`() {
+    val movementWithoutPnc = movementFactory(prisonNumber = "testPrisonNumber", pncNumber = null)
+
+    every { basmService.getMoves("MDI", LocalDate.of(2020, 1, 2), LocalDate.of(2020, 1, 2)) } returns listOf(movementWithoutPnc)
+    every { prisonService.getMoves("MDI", LocalDate.of(2020, 1, 2)) } returns emptyList()
+
+    every { prisonerSearchService.matchPrisoner("testPrisonNumber") } returns listOf(MatchPrisonerResponse("testPrisonNumber", "testPncNumber"))
+
+    val moves = movementService.getMovementsMatchedWithPrisoner("MDI", LocalDate.of(2020, 1, 2))
+
+    verify(exactly = 1) { prisonerSearchService.matchPrisoner("testPrisonNumber") }
+    assertThat(moves[0].pncNumber).isEqualTo("testPncNumber")
+  }
+
+  @Test
+  fun `getMovementsMatchedWithPrisoner - Prison Number null if no matches`() {
+    val movement = movementFactory(prisonNumber = "testPrisonNumber", pncNumber = "testPncNumber")
+
+    every { basmService.getMoves("MDI", LocalDate.of(2020, 1, 2), LocalDate.of(2020, 1, 2)) } returns listOf(movement)
+    every { prisonService.getMoves("MDI", LocalDate.of(2020, 1, 2)) } returns emptyList()
+
+    every { prisonerSearchService.matchPrisoner("testPrisonNumber") } returns emptyList()
+    every { prisonerSearchService.matchPrisoner("testPncNumber") } returns emptyList()
+
+    val moves = movementService.getMovementsMatchedWithPrisoner("MDI", LocalDate.of(2020, 1, 2))
+
+    verify(exactly = 1) { prisonerSearchService.matchPrisoner("testPrisonNumber") }
+    verify(exactly = 1) { prisonerSearchService.matchPrisoner("testPncNumber") }
+    assertThat(moves[0].prisonNumber).isEqualTo(null)
+  }
+
+  @Test
+  fun `getMovementsMatchedWithPrisoner - given Prison Number match returned and PNC Number match returned and PNC Numbers don't match between responses, Prison Number and PNC Number are null`() {
+    val movement = movementFactory(prisonNumber = "nonsensePrisonNumber", pncNumber = "nonsensePncNumber")
+
+    every { basmService.getMoves("MDI", LocalDate.of(2020, 1, 2), LocalDate.of(2020, 1, 2)) } returns listOf(movement)
+    every { prisonService.getMoves("MDI", LocalDate.of(2020, 1, 2)) } returns emptyList()
+
+    every { prisonerSearchService.matchPrisoner("nonsensePrisonNumber") } returns listOf(MatchPrisonerResponse("testPrisonNumber", "something"))
+    every { prisonerSearchService.matchPrisoner("nonsensePncNumber") } returns listOf(MatchPrisonerResponse("testPrisonNumber", "somethingDifferent"))
+
+    val moves = movementService.getMovementsMatchedWithPrisoner("MDI", LocalDate.of(2020, 1, 2))
+
+    verify(exactly = 1) { prisonerSearchService.matchPrisoner("nonsensePrisonNumber") }
+    verify(exactly = 1) { prisonerSearchService.matchPrisoner("nonsensePncNumber") }
+    assertThat(moves[0].prisonNumber).isEqualTo(null)
+    assertThat(moves[0].pncNumber).isEqualTo(null)
+  }
```

Comment:
`LocalDate.of(2020, 1, 2)` is created multiple times throughout these tests.
Suggest extracting that value to a constant and referring to that in the tests.
This will make the code more concise, easier to read and understand.